### PR TITLE
cpu: make pm_layered a DEFAULT_MODULE

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This makes `pm_layered` a `DEFAULT_MODULE` so it can be disabled.
This is useful to squeeze down the size of riotboot, which does not need pm functionality.

### Testing procedure

`riotboot` should still work.
In fact, also normal applications should still work. I tested this with `examples/default` and it will operate just normal, albeit consuming more power as it's now busy looping instead of idling when no thread is scheduled.

### Issues/PRs references

split off #15493

### CODE REVIEW

**Comment by fjmolinas** [comment]
Looks good and nice fix (I had done this on side projects well so I think it's definitely useful), I'll run the dependency check to be sure.

File: Makefile.dep [hunk 1]
```diff
@@ -86,8 +86,10 @@ endif
 # always select gpio (until explicit dependencies are sorted out)
 FEATURES_OPTIONAL += periph_gpio
 
-# always select power management if available
-FEATURES_OPTIONAL += periph_pm
+# always select power management if not building the bootloader
+ifneq (1, $(RIOTBOOT_BUILD))
+  FEATURES_OPTIONAL += periph_pm
+endif
```
Comment (fjmolinas) [inline Makefile.dep:92]: Does this save rom?

Comment (fjmolinas) [reply Makefile.dep:92]: Could you add a comment on why this is done?

Comment (benpicco) [reply Makefile.dep:93]: I expanded on the comment.

**Review by fjmolinas — APPROVED** [review]
ACK, nothing weird in the dependency script, but the fact that `periph_pm` is now required or a `default_module`.

**Comment by fjmolinas** [comment]
Please squash @benpicco !

**Pull request merged**